testapp/src/main/res/layout/item_list_miniapp.xml

```diff
@@ -41,33 +45,71 @@
             android:paddingBottom="@dimen/small_8"
             app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintStart_toEndOf="@+id/iv_app_icon"
+            app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintTop_toTopOf="parent">
 
             <TextView
                 android:id="@+id/tv_app_name"
-                android:layout_width="wrap_content"
+                android:layout_width="0dp"
                 android:layout_height="wrap_content"
                 android:textColor="@android:color/black"
                 android:textSize="@dimen/text_medium_14"
                 android:text="@{miniapp.name}"
+                android:singleLine="true"
+                android:ellipsize="end"
                 app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
                 app:layout_constraintTop_toTopOf="parent"
                 tools:text="Name of the Mini App" />
 
             <TextView
                 android:id="@+id/tv_description"
-                android:layout_width="wrap_content"
+                android:layout_width="0dp"
                 android:layout_height="wrap_content"
                 android:layout_marginTop="@dimen/small_8"
                 android:alpha="0.6"
                 android:gravity="start"
                 android:textColor="@android:color/black"
                 android:textSize="@dimen/text_medium_14"
-                android:text="@{miniapp.description}"
+                android:text="@{`Description: ` + (TextUtils.isEmpty(miniapp.description) ? `--` : miniapp.description)}"
```

Comment:
Consistency is always preferred. How about extracting description into string res like version ID or doing the other way for both? I think `:` can also be merged.

Reply:
Right. Updated